core/src/com/unciv/ui/newgamescreen/NewGameScreen.kt

```diff
@@ -98,6 +98,34 @@ class NewGameScreen(
         rightSideButton.enable()
         rightSideButton.setText("Start game!".tr())
         rightSideButton.onClick {
+            if (gameSetupInfo.gameParameters.randomNumberOfCityStates) {
+                val half = gameSetupInfo.gameParameters.numberOfCityStates / 2
+                val otherHalf = gameSetupInfo.gameParameters.numberOfCityStates - half // This is to handle odd numbers
+                gameSetupInfo.gameParameters.numberOfCityStates = half + Random().nextInt(otherHalf)
+            }
+
+            if (gameSetupInfo.gameParameters.randomNumberOfPlayers) {
+                val half = gameSetupInfo.gameParameters.players.size / 2
+                val otherHalf = gameSetupInfo.gameParameters.players.size - half // This is to handle odd numbers
```

Comment:
If I were you I'd rather let the players decide what "random" should be. 

That is:
 a `min city states` and `min major civs` sliders
and
a `max city states` and `max major civs` sliders

Reply:
I was thinking the same thing, started out that way. But it is too complicated because I'd have to manipulate the player table on the right to add and remove players as you slide the max players.

Reply:
There is also the extra validation to make sure the user chooses max larger than min.

Reply:
Maybe we could start this way and later you can upgrade it to use 4 sliders.